### PR TITLE
[NavigationDrawer] Address issue where scrim is visible when scrolled to end of content.

### PR DESCRIPTION
**Pull request by andrewoverton**
Like #8612 and #8618, this is not a beautiful solution. It does fix the issue that led to #8618 being reverted though!

### CODE REVIEW

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h [hunk 1]
```diff
@@ -24,6 +24,20 @@
  Delegate for MDCBottomDrawerContainerViewController.
  */
 @protocol MDCBottomDrawerContainerViewControllerDelegate <NSObject>
+
+/**
+This method is called when the MDCBottomDrawerContainerViewControllerDelegate needs to update the
+appearance of the scrim.
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h:30]: can you please elaborate here as in what cases there is a need to update the appearance of the scrim

**Review by yarneo — APPROVED** [review]

**Comment by randallli** [comment]
We should make sure that the PR description can stand on its own. Please copy info from the original attempt.

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m [hunk 2]
```diff
@@ -425,6 +433,25 @@ - (void)setDrawerShadowColor:(UIColor *)drawerShadowColor {
...
+        scrimShouldAdoptTrackingScrollViewBackgroundColor;
+    if ([self.delegate respondsToSelector:@selector
+                       (bottomDrawerContainerViewControllerNeedsScrimAppearanceUpdate:
+                                    scrimShouldAdoptTrackingScrollViewBackgroundColor:)]) {
+      [self.delegate bottomDrawerContainerViewControllerNeedsScrimAppearanceUpdate:self
+                                 scrimShouldAdoptTrackingScrollViewBackgroundColor:
+                                     _scrimShouldAdoptTrackingScrollViewBackgroundColor];
+    }
+  }
+  self.shadowedView.layer.shadowColor = _scrimShouldAdoptTrackingScrollViewBackgroundColor
```
Comment (randallli) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:451]: lets mention the bug in a comment above this line here too.
    // This logic is to mitigate b/119714330.

**Review by randallli — APPROVED** [review]

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h (hunk 1 above)
Comment (andrewoverton) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h:30]: I'm not an expert on this component, so I'm approaching this problem empirically, i.e. just seeing what works, and it seems that these are the times where the scrim (and now `shadowedView.layer.shadowColor`) should not be visible. There are three variables at play:
1) Whether the header has reached the top of the screen/window/whatever it's in.
2) Whether the end of the trackingScrollView's content has been reached.
3) Whether there is enough content in the trackingScrollView to exceed the screen height.

The interplay between these three things results in this variable needing to be set to this value during these specific times. I'm kind of in a hurry right now, but I can come back later and add annotations to each logical branch that might help elucidate this interplay if you'd like?

**Pull request merged**